HPC-8351: Upgrade Node.js from v12 to v16

Pl217 opened the pull request:
Needed for https://github.com/UN-OCHA/hpc_service/pull/2517 to pass the CI, because CDM packages are first installed and then hpc_service packages. CI fails if legacy lockfile version (version 1) is used. Both `npm` v6 and v7+ CLIs can read both lockfile version 1 and lockfile version 2, but make their respective modifications to them, causing CI to fail.

Pl217 committed:
HPC-8351: Upgrade Node.js from v12 to v16 in Docker image

Pl217 committed:
HPC-8351: Require npm v8

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

Pl217 committed:
HPC-8351: Require Node.js v16 in package.json and be strict about it

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

Pl217 committed:
HPC-8351: Be explicit about Node version in GitHub CI

Even though Node v16 is the default
https://github.com/actions/virtual-environments/issues/4446
be explicit about the version needed, because
future updates may break CI

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

Pl217 committed:
⬆️ Upgrade `@types/node` to `v16.11.21`

Pl217 committed:
➖ Uninstall `ts-node` dependency

This dependency was there since the initial commit
83c811c967d3de6ff2d63f690d58752af0a85c8b, but seems to
have never been used.

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

Pl217 committed:
Format GitHub YML config files

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review

unocha-hpc reviewed (approved):
Checks have passed and this pull request is ready for manual review
